main-app/app/scripts/bingoNumbers/callingNumbers.js

```diff
@@ -0,0 +1,45 @@
+(function () {
+    'use strict';
+    angular.module('Tombola.LongPollingService')
+        .service('longPolling',['$timeout', 'proxy', 'userModel', 'numberService', 'tokenService', function ($timeout, proxy, userModel, numberService, tokenService) {
+            var me = this;
+            me.currentNumber = 0;
+            me.startNumber = 1;
+            me.winnerMessage ='';
+            me.numbersToWin = 5;
+            me.winnerHasNotBeenFound = true;
+
+            me.startPolling = function() {
+                proxy.getBingoNumber(tokenService.getToken(), userModel.name, userModel.currentBalance, me.startNumber).then(function(response) {
+                    me.currentNumber = response.ball;
+                    me.startNumber += 1;
+                    checkForWinnerfound(response);
+                    numberService.findNumberInTicket(me.currentNumber);
+                    if(me.winnerHasNotBeenFound){
+                        $timeout(me.startPolling, 500);
```

Comment:
Don't recursively call - just use a $interval and cancel when won. You can set a count to act as a backup in case the server doesn't provide the house prize 
https://docs.angularjs.org/api/ng/service/$interval